python/pyspark/tests/test_rdd.py

```diff
@@ -726,6 +729,13 @@ def stopit(*x):
         self.assertRaisesRegexp((Py4JJavaError, RuntimeError), msg,
                                 seq_rdd.aggregate, 0, lambda *x: 1, stopit)
 
+    def test_overwritten_global_func(self):
+        # Regression test for SPARK-27000
+        global global_func
+        self.assertEqual(self.sc.parallelize([1]).map(lambda _: global_func()).first(), "Hi")
+        global_func = lambda: "Yeah"
+        self.assertEqual(self.sc.parallelize([1]).map(lambda _: global_func()).first(), "Yeah")
```

Comment:
I ran a test locally:

```python
>>> global_func = lambda: "Hi"
>>> sc.parallelize([1]).map(lambda _: global_func()).first()
'Hi'
>>> global_func = lambda: "Yeah"
>>> sc.parallelize([1]).map(lambda _: global_func()).first()
'Hi'
>>> sc.parallelize([1]).map(lambda _: global_func()).first()
'Yeah'
>>> sc.parallelize([1]).map(lambda _: global_func()).first()
'Hi'
```

Seems it outputs `Hi` or `Yeah` randomly. Is it caused by this cloudpickle issue?

Reply:
Yes, it looks so. After the fix, the result becomes consistent:

```python
>>> global_func = lambda: "Hi"
>>> sc.parallelize([1]).map(lambda _: global_func()).first()
'Hi'
>>> global_func = lambda: "Yeah"
>>> sc.parallelize([1]).map(lambda _: global_func()).first()
'Yeah'
>>> sc.parallelize([1]).map(lambda _: global_func()).first()
'Yeah'
>>> sc.parallelize([1]).map(lambda _: global_func()).first()
'Yeah'

```